From 5999a83bd04d1b1f575279e0f90cc0dfc78861e7 Mon Sep 17 00:00:00 2001
From: Konstantin Chukharev <lipen00@gmail.com>
Date: Wed, 19 Jul 2023 17:32:24 +0300
Subject: [PATCH] Add getc_unlocked on Windows

---
 drat-trim.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drat-trim.c b/drat-trim.c
index a8ad52e..f590742 100644
--- a/drat-trim.c
+++ b/drat-trim.c
@@ -55,6 +55,11 @@ OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWA
 
 #define COMPRESS
 
+#ifdef _WIN32
+int getchar_unlocked() { return _getchar_nolock(); }
+int getc_unlocked(FILE *stream) { return _getc_nolock(stream); }
+#endif
+
 struct solver { FILE *inputFile, *proofFile, *lratFile, *traceFile, *activeFile;
     int *DB, nVars, timeout, mask, delete, *falseStack, *falseA, *forced, binMode, optimize, binOutput,
       *processed, *assigned, count, *used, *max, COREcount, RATmode, RATcount, nActive, *lratTable,
-- 
2.41.0.windows.1

